src/test/java/com/fasterxml/jackson/databind/struct/TestUnwrappedWithDynamicPrefix.java

```diff
@@ -0,0 +1,205 @@
+package com.fasterxml.jackson.databind.struct;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonUnwrapped;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.BeanProperty;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.deser.ContextualDeserializer;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.ser.ContextualSerializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.databind.testutil.DatabindTestUtil;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class TestUnwrappedWithDynamicPrefix extends DatabindTestUtil {
+    private final ObjectMapper MAPPER = newJsonMapper();
+    private final ObjectMapper CUSTOM_MAPPER = newJsonMapper()
+            .registerModule(new SimpleModule()
+                    .addSerializer(ValueAndMap.class, new ValueAndMapSerializer())
+                    .addDeserializer(ValueAndMap.class, new ValueAndMapDeserializer()));
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    static class TwoUnwrappedPropertiesWithSameClass {
+        @JsonUnwrapped
+        public ValueAndMap prop1;
+        @JsonUnwrapped
+        public ValueAndMap prop2;
+    }
+
+    static class ValueAndMap {
+        public String value;
+        public Map<String, String> value_l10n;
+
+        @JsonIgnore
+        public boolean isEmpty() {
+            return value == null && (value_l10n == null || value_l10n.isEmpty());
+        }
+
+    }
+
+    @Test
+    public void testUnwrappingWithDifferentProperties() throws JsonProcessingException {
+        TwoUnwrappedPropertiesWithSameClass o = new TwoUnwrappedPropertiesWithSameClass();
+        o.prop1 = new ValueAndMap();
+        o.prop1.value = "foo";
+        o.prop1.value_l10n = Map.of("a", "b");
+        o.prop2 = new ValueAndMap();
+        o.prop2.value = "bar";
+        o.prop2.value_l10n = Map.of("c", "d");
+
+        // currently produces JSON object with duplicate keys JSON
```

Comment:
I guess ideally it'd fail with exception? Since definition would instruct output like this.